packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts

```diff
@@ -54,15 +54,23 @@ function processRequest(request: RequestCompleteEvent) {
         status_code: request.status,
         url: request.url,
       },
-      type: RumEventType.RESOURCE,
+      type: RumEventType.RESOURCE as const,
     },
     tracingInfo,
     correspondingTimingOverrides
   )
-  return { startTime: startClocks.relative, rawRumEvent: resourceEvent as RawRumResourceEvent }
+  return {
+    startTime: startClocks.relative,
+    rawRumEvent: resourceEvent,
+    domainContext: {
+      performanceEntry: matchingTiming as PerformanceEntry | undefined,
```

Comment:
May I ask, why it's likely to be the `toJSON` representation?

Reply:
You can see the difference with the following on any page:

```ts
console.log(performance.getEntriesByType('resource')[0]):
console.log(performance.getEntriesByType('resource')[0].toJSON()):
```

I think the two noticeable things are:

* the `toJSON` method is not in the `toJSON` representation
* `entry instanceof PerformanceEntry` will return false

Reply:
Then I would agree on having another type called "JsonPerformanceEntry" or "PerformanceEntryRepresentation".

Reply:
I think it could do the trick (in fact, I tried that in a previous version of the PR), but I wanted to have your opinion first. I'm a bit mitigated because I don't think it is our job to provide native browser types (that's a bit cumbersome to maintain).

We could re-use our [custom `PerformanceEntry` types](https://github.com/DataDog/browser-sdk/blob/ca9e9d3998d990fbc0e5829b3318335fc855fd80/packages/rum-core/src/browser/performanceCollection.ts#L20-L85) types and expose them publicly (we introduced them to ease the mock of those objects, as well as adding the possibility to create custom ones like `initial_document` resources). But we'll need to consolidate those types (there is a few missing properties), and chose what to do for the unofficial `traceId` prop)